ldap/servers/slapd/daemon.c

```diff
@@ -2845,18 +2846,30 @@ catch_signals()
 #endif /* HPUX */
 
 static int
-get_configured_connection_table_size(void)
+get_connection_table_size(void)
 {
-    int size = config_get_conntablesize();
+    int size = 0;
+    int resrvdesc = 0;
     int maxdesc = config_get_maxdescriptors();
 
-    /*
-     * Cap the table size at nsslapd-maxdescriptors.
-     */
-    if (maxdesc >= 0 && size > maxdesc) {
-        size = maxdesc;
+    /* Validate configured reserve descriptors */
+    validate_num_config_reservedescriptors();
+
+    resrvdesc = config_get_reservedescriptors();
+    if (maxdesc > resrvdesc) {
+         size = (maxdesc - resrvdesc);
     }
 
+    /* Verify size does not exceed process max fds */
+    if (size > FDS_PROCESS_MAX) {
```

Comment:
That seems a bit suspicious as maxdesc is the process max fds.
There is a second limit related to the number of tcpip port that can be open (in therory 65536 -2) in practice I would say 65000 (too keep ssh and other tools connectivity), but it is an hard limit (resrvdesc does not matter at this point)
i.e: it should be if (size > MAX_TCP_PORTS) { size = MAX_TCP_PORTS; }

Reply:
We used FDS_PROCESS_MAX (64000) in the initial PR, but if you like I can change it to your proposal.

Reply:
IMHO you should change the name: FDS_PROCESS_MAX is misleading as you assumed it was file descriptors but
At this point we are no more dealing with the number of file descriptors as it has already been taken in account by  
 maxdesc, we are dealing with the maximum number of connections that may be open, (IMHO the right name is probably MAX_LDAP_CONNS)  and it is an hard limit. Now 64000 or 65000 may be argued, so let use the existing value.